File: app/src/main/java/com/mutant/godutch/model/Group.java
package com.mutant.godutch.model;

import java.util.List;

/**
 * Created by evanfang102 on 2017/3/30.
 */

public class Group {

    private String id;
    private String title;
    private String description;
    private String photoUrl;
    private int subtotal;
    private List<Friend> friends;

    public Group() {
    }

    public Group(String id, String title, String description, String photoUrl, int subtotal, List<Friend> friends) {
        this.id = id;
        this.title = title;
        this.description = description;
        this.photoUrl = photoUrl;
        this.subtotal = subtotal;
        this.friends = friends;
    }

    public Group(String title, String description, String photoUrl, int total, List<Friend> friends) {
        this(null, title, description, photoUrl, total, friends);
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }

    public int getSubtotal() {
        return subtotal;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public List<Friend> getFriends() {
        return friends;
    }

    public String getPhotoUrl() {
        return photoUrl;
    }

    public void setPhotoUrl(String photoUrl) {
        this.photoUrl = photoUrl;
    }
}

File: app/src/main/java/com/mutant/godutch/model/Group.kt
package com.mutant.godutch.model

import android.os.Parcel
import android.os.Parcelable
import com.firebase.client.ServerValue
import com.google.firebase.database.Exclude

/**
 * Created by evanfang102 on 2017/3/30.
 */

class Group : Parcelable {

    var title: String = ""
    var photoUrl: String = ""
    var subtotal: Int = 0
    var friendsUid: List<String> = arrayListOf()
    val timestamp: HashMap<String, Any> = hashMapOf()
    var timestampCreated: Long = 0
        @Exclude
        get() = timestamp["timestamp"] as Long? ?: 0

    @Exclude
    var key: String = ""

    constructor()

    constructor(title: String, photoUrl: String, subtotal: Int, friendsUid: List<String>) {
        this.title = title
        this.photoUrl = photoUrl
        this.subtotal = subtotal
        this.friendsUid = friendsUid
        this.timestamp.put("timestamp", ServerValue.TIMESTAMP)
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(title)
        parcel.writeString(photoUrl)
        parcel.writeInt(subtotal)
        parcel.writeStringList(friendsUid)
        parcel.writeString(key)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Group> {
        override fun createFromParcel(parcel: Parcel): Group {
            return Group(parcel)
        }

        override fun newArray(size: Int): Array<Group?> {
            return arrayOfNulls(size)
        }
    }

    constructor(parcel: Parcel) : this() {
        title = parcel.readString()
        photoUrl = parcel.readString()
        subtotal = parcel.readInt()
        friendsUid = parcel.createStringArrayList()
        key = parcel.readString()
    }

}
